### PR TITLE
syncing tokens and created contracts/tokens

### PR DESCRIPTION
**Pull request by benny-conn**
Changes:

- **Added** token syncing and creator token/contract syncing for a new chain, `ChainZora`, in a new `*zora.Provider` multichain provider
- **Added** `Zora` to graphQL
- **Added** address mapping of eth address to zora chain so that you can use any evm address to refresh zora tokens

### CODE REVIEW

File: service/multichain/zora/zora.go [hunk 1]
```diff
@@ -0,0 +1,476 @@
...
+	greq.Var("after", endCursor)
+	greq.Var("limit", limit)
+
+	resp := getTokensResponse{}
+	err := d.zgql.Run(ctx, greq, &resp)
+	if err != nil {
+		return nil, nil, fmt.Errorf("error getting tokens from zora: %w", err)
+	}
+
+	if offset > 0 {
```
Comment (jarrel-b) [inline service/multichain/zora/zora.go:154]: Should this be applied at the end after we've fetched all tokens that match the query? Could be mistaken, but It seems that this is subsetting each page instead

Somewhat related, It looks `GetTokensByWalletAddress` is always called with a limit of 0 and offset of 0. I thought we needed it for refreshing communities, but if we don't does it make sense to have providers ignore those inputs now?

**Review by jarrel-b — APPROVED** [review]
Looks awesome, just had a few comments / questions!

File: service/persist/token.go [hunk 2]
```diff
@@ -72,10 +72,12 @@ const (
...
+	// ChainZora represents the Zora blockchain
+	ChainZora
 
 	// MaxChainValue is the highest valid chain value, and should always be updated to
 	// point to the most recently added chain type.
-	MaxChainValue = ChainPOAP
+	MaxChainValue = ChainZora
 )
 
 var AllChains = []Chain{ChainETH, ChainArbitrum, ChainPolygon, ChainOptimism, ChainTezos, ChainPOAP}
```
[the remark this answers is not in the document]
Comment (benny-conn) [reply service/persist/token.go:83]: Good catch!

File: server/server.go [hunk 3]
```diff
@@ -221,6 +221,7 @@ func SetDefaults() {
 	viper.SetDefault("INFURA_API_KEY", "")
 	viper.SetDefault("INFURA_API_SECRET", "")
 	viper.SetDefault("PUSH_NOTIFICATIONS_SECRET", "push-notifications-secret")
+	viper.SetDefault("ZORA_API_KEY", "")
```
[the remark this answers is not in the document]
Comment (benny-conn) [reply server/server.go:224]: Yes, but we need to get an API key first from Zora

File: service/multichain/zora/zora.go [hunk 4]
```diff
@@ -0,0 +1,476 @@
...
+
+	nodesToTokens, _ := util.Map(resp.Tokens.Nodes, func(n tokenNode) (token, error) {
+		return n.Token, nil
+	})
+	tokens, contracts, err := d.tokensToChainAgnostic(ctx, nodesToTokens)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if resp.Tokens.PageInfo.hasNextPage {
```
[the remark this answers is not in the document]
Comment (benny-conn) [reply service/multichain/zora/zora.go:172]: Great catch!

File: service/multichain/zora/zora.go (hunk 1 above)
Comment (benny-conn) [reply service/multichain/zora/zora.go:154]: True, in every current use case we hard code 0 and 0 I believe, I guess this was a bet that we might want it in a future where hundreds of thousands of NFTs could be owned by a user maybe, but I guess we can plan for that then.

**Pull request merged**